src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -17,6 +17,17 @@ public static class ConfigurationExtensions
         private const string Secrets_File_Name = "secrets.json";
 
         /// <summary>
+        /// Adds the user secrets configuration source. Searches the assembly that contains type <typeparamref name="TStartup"/>
+        /// for an instance of <see cref="UserSecretsIdAttribute"/>.
+        /// </summary>
+        /// <param name="configuration"></param>
+        /// <typeparam name="TStartup">The startup class</typeparam>
+        /// <returns></returns>
+        public static IConfigurationBuilder AddUserSecrets<TStartup>(this IConfigurationBuilder configuration)
```

Comment:
I don't feel strongly about the method either. But the thinking was that we would switch templates to use this. `.AddUserSecrets<Startup>()` seemed cleaner than `.AddUserSecrets(typeof(Startup).GetTypeInfo().Assembly)`. 

FWIW as long as we're consider API for the sake of templates, may we should just template `.AddUserSecrets(string userSecretId)` instead.